src/util/std_code.cpp

```diff
@@ -167,3 +170,57 @@ void code_function_bodyt::set_parameter_identifiers(
     sub.back().set(ID_identifier, id);
   }
 }
+
+code_inputt::code_inputt(
+  std::vector<exprt> arguments,
+  optionalt<source_locationt> location)
+  : codet{ID_input, std::move(arguments)}
+{
+  if(location)
+    add_source_location() = std::move(*location);
+}
+
+code_inputt::code_inputt(
+  const irep_idt &identifier,
+  exprt expression,
```

Comment:
`identifier` usually is an index into the symbol table for us -- how about making that "description"?

Reply:
Sounds good to me, I will update this.

Reply:
Done.